aws/data_source_aws_route53_resolver_endpoint_test.go

```diff
@@ -0,0 +1,217 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceAwsRoute53ResolverEndpoint_Basic(t *testing.T) {
+	name := acctest.RandomWithPrefix("tf-acc-test")
+	rInt := acctest.RandInt()
+	direction := "INBOUND"
+	resourceName := "aws_route53_resolver_endpoint.foo"
+	datasourceName := "data.aws_route53_resolver_endpoint.foo"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsRoute53ResolverEndpointConfig_NonExistent,
+				ExpectError: regexp.MustCompile(`The ID provided could not be found`),
+			},
+			{
+				Config: testAccDataSourceRoute53ResolverEndpointConfig_initial(rInt, direction, name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "name", resourceName, "name"),
+					resource.TestCheckResourceAttrPair(datasourceName, "id", resourceName, "id"),
+					resource.TestCheckResourceAttr(datasourceName, "ip_addresses.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsRoute53ResolverEndpoint_Filter(t *testing.T) {
+	name := acctest.RandomWithPrefix("tf-acc-test")
+	rInt := acctest.RandInt()
+	direction := "OUTBOUND"
+	resourceName := "aws_route53_resolver_endpoint.foo"
+	datasourceName := "data.aws_route53_resolver_endpoint.foo"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsRoute53ResolverEndpointConfig_NonExistentFilter,
+				ExpectError: regexp.MustCompile("Your query returned no results. Please change your search criteria and try again"),
+			},
+			{
+				Config: testAccDataSourceRoute53ResolverEndpointConfig_filter(rInt, direction, name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "name", resourceName, "name"),
+					resource.TestCheckResourceAttrPair(datasourceName, "id", resourceName, "id"),
+					resource.TestCheckResourceAttr(datasourceName, "ip_addresses.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceRoute53ResolverEndpointConfig_base(rInt int) string {
+	return fmt.Sprintf(`
+resource "aws_vpc" "foo" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+
+  tags = {
+    Name = "terraform-testacc-r53-resolver-vpc-%d"
+  }
+}
+
+data "aws_availability_zones" "available" {}
```

Comment:
Use `testAccAvailableAZsNoOptInConfig`

https://github.com/terraform-providers/terraform-provider-aws/blob/aff071d49120c136dcb4917133b4eaa6512f0482/aws/resource_aws_instance_test.go#L3301-L3312

otherwise the default acceptance test regions may use Local Zone AZs and get errors like

```
=== CONT  TestAccDataSourceAwsRoute53ResolverEndpoint_Basic
    data_source_aws_route53_resolver_endpoint_test.go:19: Step 2/2 error: Error running apply: 2020/10/29 10:40:11 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.10.0
        
        Error: error creating Route53 Resolver endpoint: InvalidRequestException: [RSLVR-00419] There are no resolver resources in the Availability Zone containing subnet: "subnet-055bec065a57e028c"
        
        
--- FAIL: TestAccDataSourceAwsRoute53ResolverEndpoint_Basic (23.32s)
```

Reply:
Thanks @ewbankkit I'll get these resolved today.